### PR TITLE
Enable .NET 8 for uwp MultiTarget, upgrade packages

### PR DESCRIPTION
**Pull request by Arlodotexe**
For components that declare a `uwp` MultiTarget, this PR:
- Introduces the `net8.0-windows10.0.26100.0` TFM alongside the existing `uap10.0.17763` 
- Auto-enables the required properties for targeting modern dotnet in uwp
- Updates several platform packages to versions that support modern dotnet on uwp
- Adds several workarounds and fixes to get builds working again with all the latest tooling

Prerequisite PRs:
- https://github.com/CommunityToolkit/Tooling-Windows-Submodule/pull/235
- https://github.com/CommunityToolkit/Windows/pull/563

### CODE REVIEW

**Comment by Arlodotexe** [comment]
Ran into an analyzer error in the Extensions component that needs additional clarification to resolve.

#### Problem
The build is failing with the following error:
```
error CS0703: Inconsistent accessibility: constraint type 'IPredicate<T>' is less accessible than 'FrameworkElementExtensions.FindChild<T, TPredicate>(FrameworkElement, ref TPredicate)'
```
This error indicates that the `IPredicate<T>` interface has a more restrictive accessibility level than the `FindChild<T, TPredicate>` method, causing CS0703.

#### Affected Code
The affected method is `FindChild<T, TPredicate>` in the FrameworkElementExtensions.LogicalTree.cs file:

https://github.com/CommunityToolkit/Windows/blob/6392784b4ccbe2eea8295ab140d803c26bda8eed/components/Extensions/src/Tree/FrameworkElementExtensions.LogicalTree.cs#L99

While `IPredicate<T>` resides in:
https://github.com/CommunityToolkit/Windows/blob/6392784b4ccbe2eea8295ab140d803c26bda8eed/components/Extensions/src/Tree/Predicates/Interfaces/IPredicate%7BT%7D.cs#L11

#### Possible Solutions
1. **Make `IPredicate<T>` Public**: We could update the `IPredicate<T>` interface to be declared as `public` to match the accessibility level of the `FindChild<T, TPredicate>` method.
2. **Change Method Accessibility**: If the static `FindChild<T, TPredicate>` method is not intended to be used publicly, we could change its accessibility level to match that of `IPredicate<T>`.

#### Request for Clarification
What are the intended accessibility levels for the `IPredicate<T>` interface and the `FindChild<T, TPredicate>` method? Should `IPredicate<T>` be made public, or should the method's accessibility be restricted?

**Review by michael-hawker — APPROVED** [review]

**Comment by michael-hawker** [comment]
Should be gtg once the tooling pointer is updated

File: components/Extensions/src/Text/StringExtensions.Localization.cs [hunk 1]
```diff
@@ -49,16 +49,16 @@ static StringExtensions()
...
     /// <returns>string value for given resource or empty string if not found.</returns>
     ////[SupportedOSPlatform("Windows10.0.18362.0")]
-    public static string GetViewLocalized(this string resourceKey, UIContext? uiContext = null)
+    public static string? GetViewLocalized(this string resourceKey, UIContext? uiContext = null)
     {
         if (uiContext != null)
         {
             var resourceLoader = ResourceLoader.GetForUIContext(uiContext);
-            return resourceLoader.GetString(resourceKey);
+            return resourceLoader?.GetString(resourceKey) ?? string.Empty;
```
Comment (Sergio0694) [inline components/Extensions/src/Text/StringExtensions.Localization.cs:57]: There shouldn't be a null check here. Also the method already returns nullable, we shouldn't do `?? string.Empty`
```suggestion
            return resourceLoader.GetString(resourceKey);
```
Same for the other branch below.

**Review by Sergio0694 — APPROVED** [review]

File: components/Converters/src/ResourceNameToResourceStringConverter.cs [hunk 2]
```diff
@@ -32,17 +32,18 @@ public sealed partial class ResourceNameToResourceStringConverter : IValueConver
...
         {
             return string.Empty;
         }
 
 #if WINAPPSDK && !HAS_UNO
-        return _resourceManager.MainResourceMap.TryGetValue(value.ToString()).ValueAsString;
+        return _resourceManager.MainResourceMap.TryGetValue(valueAsString).ValueAsString;
 #else
-        return _resourceLoader.GetString(value.ToString());
+        return _resourceLoader.GetString(valueAsString) ?? string.Empty;
```
Comment (Sergio0694) [inline components/Converters/src/ResourceNameToResourceStringConverter.cs:46]:
```suggestion
        return _resourceLoader.GetString(valueAsString);
```

File: components/Extensions/src/Text/StringExtensions.Localization.cs [hunk 3]
```diff
@@ -58,7 +58,7 @@ public static string GetViewLocalized(this string resourceKey, UIContext? uiCont
         }
         else
         {
-            return ResourceLoader.GetForCurrentView().GetString(resourceKey);
+            return ResourceLoader.GetForCurrentView().GetString(resourceKey) ?? string.Empty;
```
Comment (Sergio0694) [inline components/Extensions/src/Text/StringExtensions.Localization.cs:61]:
```suggestion
            return ResourceLoader.GetForCurrentView().GetString(resourceKey);
```

**Review by Sergio0694 — CHANGES_REQUESTED** [review]

**Review by Sergio0694 — APPROVED** [review]
:shipit:

File: nuget.config [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <configuration>
   <packageSources>
-    <clear />
     <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+    <add key="PullRequests" value="https://pkgs.dev.azure.com/dotnet/CommunityToolkit/_packaging/CommunityToolkit-PullRequests/nuget/v3/index.json" protocolVersion="3" />
```
Comment (michael-hawker) [inline nuget.config:4]: I thought this was reverted? This is causing the packaging step error, we shouldn't have this here.

File: nuget.config [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <configuration>
   <packageSources>
-    <clear />
```
Comment (michael-hawker) [inline nuget.config:3]: This needs to be added back

**Review by michael-hawker — CHANGES_REQUESTED** [review]
Nuget.config changes need to be reverted

File: nuget.config (hunk 4 above)
Comment (Arlodotexe) [reply nuget.config:4]: Good catch, this shouldn't be here anymore.

**Review by michael-hawker — APPROVED** [review]

**Pull request merged**